File: java/000/ListNode.java

public class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }

    public static void main(String[] args) {
        printListNode(getListNode(1, 5, 6, 2), getListNode(7, 8, 6, 2));
    }

    public static ListNode getListNode(int... is) {
        ListNode re = new ListNode(is[0]), prev = re;
        for (int i = 1; i < is.length; i++)
            prev = prev.next = new ListNode(is[i]);
        return re;
    }

    public static void printListNode(ListNode... lns) {
        for (ListNode ln : lns) {
            while (ln != null) {
                System.out.print(ln.val + "\t");
                ln = ln.next;
            }
            System.out.print('\n');
        }
    }
}

File: kt/000/ListNode.kt

fun main(args: Array<String>) {
    ListNode.printListNode(ListNode.getListNode(1, 5, 6, 2), ListNode.getListNode(7, 8, 6, 2))
}

class ListNode(var `val`: Int = 0) {
    var next: ListNode? = null

    companion object {
        fun getListNode(vararg `is`: Int): ListNode {
            val re = ListNode(`is`[0])
            var prev = re
            for (i in 1 until `is`.size) {
                prev.next = ListNode(`is`[i])
                prev = prev.next!!
            }
            return re
        }

        fun printListNode(vararg lns: ListNode?) {
            for (ln in lns) {
                var tmp :ListNode?= ln
                while (tmp != null) {
                    print(tmp.`val`.toString() + "\t")
                    tmp = tmp.next
                }
                print('\n')
            }
        }
    }
}
